src/explorer/helpers/node-icon.ts

```diff
@@ -1,12 +1,30 @@
 import { NodeType } from "~app/types"
 import fileImage from "~images/file.png"
+import textFileImage from "~images/text_file.png"
+import audioFileImage from "~images/audio_file.png"
 import folderImage from "~images/folder.png"
+import fullFolderImage from "~images/full_folder.png"
 import unknownImage from "~images/unknown.png"
 
-export const nodeIcon = (type: NodeType) => {
+export const nodeIcon = (type: NodeType, name: string) => {
+  if (type === NodeType.FOLDER || type === NodeType.FULL_FOLDER) {
+    return getFolderIcon(type)
+  } else {
+    // TODO : double check this
+    const extension = name.split('.').pop();
```

Comment:
as we have a default in the switch, I think this works just fine and won't break if for example we don't actually have an extension